File: app/src/main/java/com/gmail/fattazzo/formula1world/domain/F1Driver.java
package com.gmail.fattazzo.formula1world.domain;

import org.apache.commons.lang3.StringUtils;

import java.io.Serializable;
import java.util.Date;

/**
 * @author fattazzo
 *         <p/>
 *         date: 05/06/17
 */
public class F1Driver implements Serializable {

    public String driverRef;

    public int number;

    public String code;

    public String url;

    public String givenName;
    public String familyName;

    public Date dateOfBirth;

    public String nationality;

    public String getFullName() {
        return StringUtils.defaultString(givenName) + " " + StringUtils.defaultString(familyName);
    }

    @Override
    public String toString() {
        return "Driver{" +
                "driverRef='" + driverRef + '\'' +
                ", number=" + number +
                ", code='" + code + '\'' +
                ", url='" + url + '\'' +
                ", givenName='" + givenName + '\'' +
                ", familyName='" + familyName + '\'' +
                ", dateOfBirth='" + dateOfBirth + '\'' +
                ", nationality='" + nationality + '\'' +
                '}';
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        F1Driver driver = (F1Driver) o;

        return driverRef.equals(driver.driverRef);

    }

    @Override
    public int hashCode() {
        return driverRef.hashCode();
    }
}

File: app/src/main/kotlin/com/gmail/fattazzo/formula1world/domain/F1Driver.kt
/*
 * Project: total-gp-world
 * File: F1Driver.kt
 *
 * Created by fattazzo
 * Copyright © 2018 Gianluca Fattarsi. All rights reserved.
 *
 * MIT License
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

package com.gmail.fattazzo.formula1world.domain

import org.apache.commons.lang3.StringUtils
import java.io.Serializable
import java.util.*

/**
 * @author fattazzo
 *
 *
 * date: 05/06/17
 */
class F1Driver : Serializable {

    var driverRef: String? = null

    var number: Int = 0

    var code: String? = null

    var url: String? = null

    var givenName: String? = null
    var familyName: String? = null

    var dateOfBirth: Date? = null

    var nationality: String? = null

    val fullName: String
        get() = StringUtils.defaultString(givenName) + " " + StringUtils.defaultString(familyName)

    override fun toString(): String {
        return "Driver{" +
                "driverRef='" + driverRef + '\'' +
                ", number=" + number +
                ", code='" + code + '\'' +
                ", url='" + url + '\'' +
                ", givenName='" + givenName + '\'' +
                ", familyName='" + familyName + '\'' +
                ", dateOfBirth='" + dateOfBirth + '\'' +
                ", nationality='" + nationality + '\'' +
                '}'
    }

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false

        val driver = o as F1Driver?

        return driverRef == driver!!.driverRef

    }

    override fun hashCode(): Int {
        return driverRef!!.hashCode()
    }
}
